packages/pyright-internal/src/analyzer/typeEvaluator.ts

```diff
@@ -3016,16 +3016,19 @@ export function createTypeEvaluator(importLookup: ImportLookup, evaluatorOptions
         });
 
         if (!diagAddendum.isEmpty()) {
+            const simple = diagAddendum.getChildren().length === 1 && isSimpleMessageStyle();
             const fileInfo = AnalyzerNodeInfo.getFileInfo(target);
             addDiagnostic(
                 fileInfo.diagnosticRuleSet.reportGeneralTypeIssues,
                 DiagnosticRule.reportGeneralTypeIssues,
-                (target.nodeType === ParseNodeType.List
-                    ? Localizer.Diagnostic.listAssignmentMismatch()
-                    : Localizer.Diagnostic.tupleAssignmentMismatch()
-                ).format({
-                    type: printType(type),
-                }) + diagAddendum.getString(),
+                simple
+                    ? diagAddendum.getChildren()[0].getChildren()[0].getString().trim()
```

Comment:
This is a bit nasty but a fairly low impact intervention to improve the message for our usecase.
The second nested addendum is the key piece of information.